### PR TITLE
Add compiler-specific search directories in getResolvedLibraryPath

### PR DESCRIPTION
**Pull request by hainest**
This extends the functionality of #731 to automatically find and search compiler-specific library paths when using gcc and clang.

### CODE REVIEW

**Comment by hainest** [comment]
@mxz297 @sashanicolas Do you know if there are already some macros to do macro stringification somewhere in the codebase?

**Comment by hainest** [comment]
https://bottle.cs.wisc.edu/search?dyninst_branch=PR732

This has fixed the static rewriter issues on [coriander](https://bottle.cs.wisc.edu/details/642) without user intervention.

**Comment by mxz297** [comment]
@hainest I am not aware of such code.

**Comment by hainest** [comment]
https://bottle.cs.wisc.edu/search?dyninst_branch=PR732&since_date=2020-04-15

This is now passing on all platforms (the regressions shown are unrelated to this PR). Importantly, `test_reloc` is now passing everywhere (modulo some intermittent failures).

@mxz297 @sashanicolas From these test results, it looks like unconditionally adding libgcc.a is not causing any issues. Do either of you have other platforms you want to test this on before accepting it?

File: dyninstAPI/CMakeLists.txt [hunk 1]
```diff
@@ -205,6 +205,23 @@ if (UNIX)
...
+	# gcc/clang search directories
+	if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+		execute_process(
+			COMMAND ${CMAKE_CXX_COMPILER} -print-search-dirs
+			OUTPUT_VARIABLE SEARCH_DIRS
+		)
+		if(SEARCH_DIRS MATCHES "libraries:[ ]?[=]?(.+)\n")
+			set(DYNINST_COMPILER_SEARCH_DIRS ${CMAKE_MATCH_1})
+		else()
+			set(DYNINST_COMPILER_SEARCH_DIRS ${CMAKE_MATCH_1})
```
[the remark this answers is not in the document]
Comment (hainest) [reply dyninstAPI/CMakeLists.txt]: I don't know why the second line was there. If we don't match, then we should just have an empty result.

**Pull request merged**